spec/2_system_overview.md

```diff
@@ -1,3 +1,51 @@
+
+## Oracle Data Request
+
+### Publishing Data Sources and Oracle Scripts
+
+Before any data requests can be made, two conditions must be met:
+
+1. The oracle script that describes the data request must also have been published
+2. The data sources related to the aforementioned oracle script must be published to Band’s system
+
+
+### Data Sources
+
+Data Sources
+Data sources are the most fundamental unit of the oracle system. It describes the procedure to retrieve a raw data point from a primary source and the fee associated with one data query. On BandChain, anyone can submit and register a data source into the system.
+To publish a data source, one must send a `MsgCreateDataSource` message to the chain. Once deployed a data source is assigned a unique `int64` identifier which can be used to refer to it forever. If the data source owner wants to edit its details once it has been published, they can use that unique identifier as a reference ID.
+
+### Oracle Scripts
+
+An oracle script is an executable program that encodes (1) the set of raw data requests to the sources it needs and (2) the way to aggregate raw data reports into the final result. These sources can be the data sources published on the network, as well as other oracle scripts. This composability makes oracle scripts very similar to [smart contracts](https://en.wikipedia.org/wiki/Smart_contract).
+
+### Oracle Data Request Initialization
+
+A data request to Band's oracle begins when an external party requests data from our oracle by broadcasting `MsgRequestData` *(TODO: LINK TO MSGREQUESTDATA DESCRIPTION)*. The contents of the message includes the oracle script the requester wants to invoke and other query and security parameters.
+
+Once the data transaction is confirmed on BandChain, the requested oracle script will begin its execution, which can be split into two phases. 
+
+### Fetching Data from Specified Data Sources
+
+First, the oracle script’s preparation function will emit the set of raw data requests necessary to continue the oracle script’s execution.The chain’s validators, who are chosen at random for security reasons, will then inspect the raw data requests and execute the associated data sources' procedures as instructed by the request. Each validator that successfully retrieved the data will then submit a raw data report to BandChain, containing the results they got from each of the sources, by broadcasting MsgReportData *(TODO: LINK TO MSGREPORTDATA DESCRIPTION)*. Once a sufficient number of validators, specified in the data request’s security parameters, have reported the raw data results, Once a sufficient number of reports are collected, this will resume BandChain’s execution of the oracle script and begin the script’s second phase of execution.
```

Comment:
> Once a sufficient number of validators, specified in the data request’s security parameters, have reported the raw data results, Once a sufficient number of reports are collected, this will resume BandChain’s execution of the oracle script and begin the script’s second phase of execution.

The first sentence is incomplete?